homeassistant/components/google_assistant/trait.py

```diff
@@ -327,6 +336,75 @@ async def execute(self, command, data, params, challenge):
         }
 
 
+@register_trait
+class ObjectDetection(_Trait):
+    """Trait to object detection.
+
+    https://developers.google.com/actions/smarthome/traits/objectdetection
+    """
+
+    name = TRAIT_OBJECTDETECTION
+    commands = []
+
+    @staticmethod
+    def supported(domain, features, device_class, _) -> bool:
+        """Test if state is supported."""
+        return domain == event.DOMAIN
+
+    def sync_attributes(self):
+        """Return ObjectDetection attributes for a sync request."""
+        return {}
+
+    def sync_options(self) -> dict[str, Any]:
+        """Add options for the sync request."""
+        return {"notificationSupportedByAgent": True}
+
+    def query_attributes(self):
+        """Return ObjectDetection query attributes."""
+
+        async def _async_send_notification(time_stamp: int) -> None:
```

Comment:
This looks too complex. It should just be a query_notifications() function which is used to build the notification data. Same as query_attributes().

Reply:
Sure, it needs to be simpeler, but my main concern for now is that triggering the notifications does not work.